theforeman.org/pipelines/lib/packaging.groovy

```diff
@@ -472,22 +472,17 @@ def rsync_debian(user, ssh_key, suite, component, deb_paths) {
     }
 }
 
-def rsync_to_yum_stage(collection, target, version) {
-    def user = 'yumrepostage'
+def rsync_to_yum_stage(collection, version) {
     def ssh_key = '/home/jenkins/workspace/staging_key/rsync_yumrepostage_key'
 
-    rsync_yum(user, ssh_key, collection, target, version)
-}
-
-def rsync_yum(user, ssh_key, collection, target, version) {
-    def hosts = ["web01.osuosl.theforeman.org"]
-
-    for(host in hosts) {
-        def target_path = "${user}@${host}:rsync_cache/${target}/${version}/"
-
-        sh """
-            export RSYNC_RSH="ssh -i ${ssh_key}"
-            /usr/bin/rsync --checksum --times --perms --recursive --links --verbose --partial --one-file-system --delete-after ${collection}/${version}/ ${target_path}
-        """
+    if (!fileExists('upload_stage_rpms')) {
+        git url: "https://github.com/theforeman/theforeman-rel-eng", poll: false
     }
```

Comment:
I know, this is contrary to what Ewoud asked for, but the previous step is always `build_stage_repository` which comes from the same repo. So I'd argue it's either guaranteed that the repo is already checked out (one could also argue to have a prep step that does only that, for clarity), or whatever is on disk should absolutely not be synced.